### PR TITLE
Spring boot data jpa

### PR DESCRIPTION
**Pull request by Yam0r**

### CODE REVIEW

File: src/main/java/mate/academy/springintro/service/BookRepository.java [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.springintro.service;
+
+import java.util.List;
+import mate.academy.springintro.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BookRepository extends JpaRepository<Book, Long> {
+    Book save(Book book);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/BookRepository.java]: remove methods

File: src/main/java/mate/academy/springintro/service/impl/BookRepositoryImpl.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package mate.academy.springintro.service.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+import mate.academy.springintro.model.Book;
+import mate.academy.springintro.service.BookRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public abstract class BookRepositoryImpl implements BookRepository {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookRepositoryImpl.java]: remove class

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package mate.academy.springintro.service.impl;
+
+import java.util.List;
+import mate.academy.springintro.model.Book;
+import mate.academy.springintro.service.BookRepository;
+import mate.academy.springintro.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java:16]:
```suggestion
@Service
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+public class BookServiceImpl implements BookService {
+
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java:18]:
```suggestion
```

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]: remove constructor

File: src/main/java/mate/academy/springintro/service/BookRepository.java [hunk 6]
```diff
@@ -0,0 +1,13 @@
+package mate.academy.springintro.service;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/BookRepository.java]:
```suggestion
package mate.academy.springintro.repository;
```

File: src/main/java/mate/academy/springintro/SpringIntroApplication.java [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+    @Autowired
+    private BookService bookService;
+
+    public static void main(String[] args) {
+        SpringApplication.run(SpringIntroApplication.class, args);
+    }
+
+    @Override
+    public void run(String... args) throws Exception {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/SpringIntroApplication.java]: remove

File: src/main/java/mate/academy/springintro/SpringIntroApplication.java [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SpringIntroApplication implements CommandLineRunner {
+
+    @Autowired
+    private BookService bookService;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/SpringIntroApplication.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/SpringIntroApplication.java [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+import java.math.BigDecimal;
+import mate.academy.springintro.model.Book;
+import mate.academy.springintro.service.BookService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+
+@SpringBootApplication
+public class SpringIntroApplication implements CommandLineRunner {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/SpringIntroApplication.java]: Where is your controller with task implementation?

Book Endpoints:

GET: /api/books (Retrieve book catalog) (should be done in the previous PR)
GET: /api/books/{id} (Retrieve book details) (should be done in the previous PR)
POST: /api/books (Create a new book) (should be done in the previous PR)
PUT: /api/books/{id} (Update a specific book)

File: src/main/resources/application.properties [hunk 10]
```diff
@@ -0,0 +1,9 @@
+spring.datasource.url=jdbc:mysql://localhost:3306/book
+spring.datasource.username=Admin
+spring.datasource.password=admin
+spring.datasource.driver-class-name=com.mysql.cj.jdbc.Driver
+spring.jpa.database-platform=org.hibernate.dialect.MySQLDialect
+spring.jpa.hibernate.ddl-auto=update
```
Comment (Elena-Bruyako) [inline src/main/resources/application.properties]:
```suggestion
spring.jpa.hibernate.ddl-auto=validate
```

File: src/test/java/mate/academy/springintro/SpringIntroApplicationTests.java [hunk 11]
```diff
@@ -0,0 +1,5 @@
+package mate.academy.springintro;
+
+public class SpringIntroApplicationTests {
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springintro/SpringIntroApplicationTests.java]:
```suggestion
@SpringBootTest
public class SpringIntroApplicationTests {
```

File: src/test/java/mate/academy/springintro/SpringIntroApplicationTests.java [hunk 12]
```diff
@@ -0,0 +1,5 @@
...
+
```
Comment (Elena-Bruyako) [inline src/test/java/mate/academy/springintro/SpringIntroApplicationTests.java:4]:
```
@Test
    void contextLoads() {
    }
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's stick to the task description:

Add Liquibase support
Change spring.jpa.hibernate.ddl-auto property value to validate
Implement soft delete concept for the Book entity
Add missing endpoints and methods for BookService

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Please, don't send new HW if you don't finish previous one. You have two open PRs with mistakes that should be fixed.

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java:19]:
```suggestion
@RestController
@RequiredArgsConstructor
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+@RequestMapping("/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java:26]: remove

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/api/books")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
@RequestMapping("/books")
```

add `server.servlet.context-path=/api` to application.properties

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @GetMapping
+    public List<Book> findAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> findById(@PathVariable Long id) {
+        Optional<Book> book = bookService.findById(id);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
        return bookService.findById(id);
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+        this.bookService = bookService;
+    }
+
+    @GetMapping
+    public List<Book> findAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> findById(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto findById(@PathVariable Long id) {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+public class BookController {
+    private final BookService bookService;
+
+    @Autowired
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+    @GetMapping
+    public List<Book> findAll() {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public List<BookDto> findAll() {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<Book> findById(@PathVariable Long id) {
+        Optional<Book> book = bookService.findById(id);
+        return book.map(ResponseEntity::ok).orElse(ResponseEntity.notFound().build());
+    }
+
+    @PostMapping
+    public Book create(@RequestBody Book book) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto create(@RequestBody CreateBookRequestDto book) {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+        return bookService.save(book);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<Book> update(@PathVariable Long id, @RequestBody Book book) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto update(@PathVariable Long id, @RequestBody CreateBookRequestDto book) {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+        return bookService.findById(id)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
        return bookService.updateBook(id, book);
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+                    existingBook.setDescription(book.getDescription());
+                    existingBook.setCoverImage(book.getCoverImage());
+                    bookService.save(existingBook);
+                    return ResponseEntity.ok(existingBook);
+                })
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> delete(@PathVariable Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public void delete(@PathVariable Long id) {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+        bookService.delete(id);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
        bookService.deleteById(id);
```

File: src/main/java/mate/academy/springintro/model/Book.java [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+    @Column(nullable = false)
+    private String author;
+
+    @Column(nullable = false, unique = true)
+    private String isbn;
+
+    @Column(nullable = false)
+    private BigDecimal price;
+
+    @Column(name = "description", columnDefinition = "TEXT")
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/Book.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/model/Book.java [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+    private String description;
+
+    private String coverImage;
+
+    @Column(nullable = false)
+    private Boolean deleted = false;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/Book.java]:
```suggestion
    private boolean isDeleted = false;
```

File: src/main/java/mate/academy/springintro/repository/BookRepository.java [hunk 26]
```diff
@@ -0,0 +1,14 @@
...
+
+import java.util.List;
+import java.util.Optional;
+import mate.academy.springintro.model.Book;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public interface BookRepository extends JpaRepository<Book, Long> {
+    List<Book> findByDeletedFalse();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/repository/BookRepository.java [hunk 27]
```diff
@@ -0,0 +1,14 @@
...
+
+    Optional<Book> findByIdAndDeletedFalse(Long id);
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/repository/BookRepository.java]:
```suggestion
```

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java (hunk 5 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]: not fixed

File: src/main/java/mate/academy/springintro/model/Book.java [hunk 28]
```diff
@@ -0,0 +1,41 @@
...
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "books")
+@Getter
+@Setter
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/Book.java:17]:
```suggestion
@SQLDelete
@SQLResrtiction 
```

???

File: src/main/java/mate/academy/springintro/repository/BookService.java [hunk 29]
```diff
@@ -0,0 +1,15 @@
+package mate.academy.springintro.repository;
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/repository/BookService.java]:
```suggestion
package mate.academy.springintro.service;
```

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 30]
```diff
@@ -0,0 +1,42 @@
...
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository;
+
+    @Autowired
+    public BookServiceImpl(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    @Override
+    public Book save(Book book) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]:
```suggestion
    public BookDto save(Book book) {
```
use Mapper

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 31]
```diff
@@ -0,0 +1,42 @@
...
+        return bookRepository.save(book);
+    }
+
+    @Override
+    public List<Book> findAll() {
+        return bookRepository.findByDeletedFalse();
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]: bookRepository.findAll
use mapper

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 32]
```diff
@@ -0,0 +1,42 @@
...
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]:
```suggestion
    public BookDto findById(Long id) {
    find or throw an exception
    use mapper
```

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 33]
```diff
@@ -0,0 +1,42 @@
...
+        return bookRepository.findByIdAndDeletedFalse(id);
+    }
+
+    @Override
+    public void delete(Long id) {
+        bookRepository.findById(id).ifPresent(book -> {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]:
```suggestion
        bookRepository.deleteById(id);
```

File: src/main/resources/db/changelog/changes/creates-book-table.yaml [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+                  constraints:
+                    nullable: false
+              - column:
+                  name: description
+                  type: text
+              - column:
+                  name: cover_image
+                  type: varchar(255)
+              - column:
+                  name: deleted
```
Comment (Elena-Bruyako) [inline src/main/resources/db/changelog/changes/creates-book-table.yaml]:
```suggestion
                  name: is_deleted
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
see pr

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 35]
```diff
@@ -0,0 +1,64 @@
...
+    @PutMapping("/{id}")
+    public ResponseEntity<BookDto> update(@PathVariable Long id, @RequestBody BookDto book) {
+        return bookService.findById(id)
+                .map(existingBook -> {
+                    existingBook.setTitle(book.getTitle());
+                    existingBook.setAuthor(book.getAuthor());
+                    existingBook.setIsbn(book.getIsbn());
+                    existingBook.setPrice(book.getPrice());
+                    existingBook.setDescription(book.getDescription());
+                    existingBook.setCoverImage(book.getCoverImage());
```
Comment (okuzan) [inline src/main/java/mate/academy/springintro/controller/BookController.java]: that should be done in mapper

**Review by okuzan — CHANGES_REQUESTED** [review]

File: src/main/java/mate/academy/springintro/controller/BookController.java (hunk 21 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/springintro/controller/BookController.java]: not fixed

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java [hunk 36]
```diff
@@ -0,0 +1,39 @@
...
+import lombok.RequiredArgsConstructor;
+import mate.academy.springintro.model.BookDto;
+import mate.academy.springintro.repository.BookRepository;
+import mate.academy.springintro.service.BookService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class BookServiceImpl implements BookService {
+    private final BookRepository bookRepository; // Внедрение репозитория
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]: remove comment

File: src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java (hunk 33 above)
Comment (Elena-Bruyako) [reply src/main/java/mate/academy/springintro/service/impl/BookServiceImpl.java]: not fixed

File: src/main/java/mate/academy/springintro/model/BookMapper.java [hunk 37]
```diff
@@ -0,0 +1,40 @@
+package mate.academy.springintro.model;
+
+import mate.academy.springintro.dto.BookRequestDto;
+import mate.academy.springintro.dto.BookResponseDto;
+import org.springframework.stereotype.Component;
+
+@Component
+public class BookMapper {
+    public BookDto toEntity(BookRequestDto dto) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/BookMapper.java]: read about MapStruct and how it works

File: src/main/java/mate/academy/springintro/model/BookDto.java [hunk 38]
```diff
@@ -0,0 +1,40 @@
...
+import javax.persistence.Id;
+import javax.persistence.Table;
+import lombok.Getter;
+import lombok.Setter;
+
+@Entity
+@Table(name = "books")
+@Getter
+@Setter
+public class BookDto {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/model/BookDto.java]: This is not DTO, it's entity

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 39]
```diff
@@ -0,0 +1,64 @@
...
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Object> delete(@PathVariable Long id) {
+        return bookService.findById(id)
+                .map(book -> {
+                    bookService.delete(id);
+                    return ResponseEntity.noContent().build();
+                })
+                .orElse(ResponseEntity.notFound().build());
+    }
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public void deleteById(@PathVariable Long id) {
        bookService.deleteById(id);
    }
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 40]
```diff
@@ -0,0 +1,64 @@
...
+
+    @PostMapping
+    public BookDto create(@RequestBody BookDto book) {
+        return bookService.save(book);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<BookDto> update(@PathVariable Long id, @RequestBody BookDto book) {
+        return bookService.findById(id)
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto updateBook(@PathVariable Long id,
                              @RequestBody CreateBookRequestDto book) {
        return bookService.updateBook(id, book);
    }
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 41]
```diff
@@ -0,0 +1,64 @@
...
+
+    @GetMapping("/{id}")
+    public ResponseEntity<BookDto> findById(@PathVariable Long id) {
+        return bookService.findById(id)
+                .map(ResponseEntity::ok)
+                .orElse(ResponseEntity.notFound().build());
+    }
+
+    @PostMapping
+    public BookDto create(@RequestBody BookDto book) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto create(@RequestBody CreateBookRequestDto book) {
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 42]
```diff
@@ -0,0 +1,64 @@
...
+    @GetMapping
+    public List<BookDto> findAll() {
+        return bookService.findAll();
+    }
+
+    @GetMapping("/{id}")
+    public ResponseEntity<BookDto> findById(@PathVariable Long id) {
+        return bookService.findById(id)
+                .map(ResponseEntity::ok)
+                .orElse(ResponseEntity.notFound().build());
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public BookDto getBookById(@PathVariable Long id) {
        return bookService.findById(id);
    }
```

File: src/main/java/mate/academy/springintro/controller/BookController.java [hunk 43]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @PostMapping
+    public BookDto create(@RequestBody BookDto book) {
+        return bookService.save(book);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<BookDto> update(@PathVariable Long id, @RequestBody BookDto book) {
```
Comment (Elena-Bruyako) [inline src/main/java/mate/academy/springintro/controller/BookController.java]:
```suggestion
    public ResponseEntity<BookDto> update(@PathVariable Long id, @RequestBody CreateBookRequestDto book) {
```

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Fix previous comments.

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's discuss you HW in chat.

**Review by sarakhmen — APPROVED** [review]

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
there are not fix comment and changes from previous HW, provide PR with only search HW

**Pull request merged**